lib/widgets/containers/tabs/dashboard_tab_widgets.dart

```diff
@@ -396,6 +406,119 @@ class SendOnSidechainViewModel extends BaseViewModel {
 
     await _router.pop();
   }
+
+  Future<void> estimateSidechainFee() async {
+    // who knows!
+    const kbyteInWithdrawal = 5;
+    final estimate = await _rpc.estimateSidechainFee(kbyteInWithdrawal);
+    sidechainFee = estimate;
+  }
+
+  void onSendOnSidechain(BuildContext context) async {
+    if (pegOutAmount == null) {
+      log.e('withdrawal amount was empty');
+      return;
+    }
+    if (sidechainFee == null) {
+      log.e('sidechain fee was empty');
+      return;
+    }
+
+    // 1. Get refund address for the sidechain withdrawal. This can be any address we control on the SC.
+    final refund = await _rpc.getRefundAddress();
+    log.d('got refund address: $refund');
+
+    final address = bitcoinAddressController.text;
+
+    // Because the function is async, the view might disappear/unmount
+    // by the time it's used. The linter doesn't like that, and wants
+    // you to check whether the view is mounted before using it
+    if (!context.mounted) {
+      return;
+    }
+
+    await showDialog(
+      context: context,
+      builder: (context) => AlertDialog(
+        title: SailText.primary14(
+          'Confirm withdrawal',
+        ),
+        content: SailText.primary14(
+          'Do you really want to execute the withdrawal?\n${bitcoinAmountController.text} BTC to $address with $sidechainFee SC fee',
+        ),
+        actions: [
+          TextButton(
+            onPressed: () => Navigator.of(context).pop(),
+            child: SailText.primary14('Cancel'),
+          ),
+          TextButton(
+            onPressed: () {
+              // Pop the currently visible dialog
+              Navigator.of(context).pop();
+
+              // This creates a new dialog on success
+              _doWithdraw(
+                context,
+                address,
+                refund,
+                pegOutAmount!,
+                sidechainFee!,
+              );
```

Comment:
Wasn't this supposed to be a sidechain internal TX?

Reply:
Yes, but how..?